### ISSUE TITLE
Standardized API for creating proxy client

### ISSUE THREAD

**Issue by mkarg** [eclipse-ee4j/jaxrs-api#598]
There are JAX-RS implementations that can build a client proxy. For example, CXF: https://cxf.apache.org/javadoc/latest/org/apache/cxf/jaxrs/client/JAXRSClientFactory.html. It is a good idea to standardize such factory APIs.

**Comment by mkarg** [comment]
Originally reported by @struberg

**Comment by mkarg** [comment]
@chkal @spericas @andymc12 What do you think?

**Comment by andymc12** [comment]
I am definitely in favor.  Personally, I'd like to see some of the work done in the MicroProfile Rest Client project migrate into the JAX-RS spec.  More info here:
http://microprofile.io/project/eclipse/microprofile-rest-client

**Comment by spericas** [comment]
I agree with @andymc12. I think we need to align this effort with the MP (See https://github.com/eclipse-ee4j/jaxrs-api/issues/56). I also think the MP Rest Client should be made part of JAX-RS. I'm not sure what's the best alias to start this discussion.

**Comment by chkal** [comment]
+1 for aligning with the MicroProfile Rest Client. Perhaps we could compare the MicroProfile client approach to the one implemented by CXF and other vendors to see if there are any fundamental differences.

**Comment by asoldano** [comment]
Speaking of client proxies... RESTEasy has http://docs.jboss.org/resteasy/docs/3.5.0.Final/userguide/html/RESTEasy_Client_Framework.html#d4e2380

**Comment by pa314159** [comment]
There is one more client proxy implementation at https://github.com/pa314159/rest-bridge that has been started some time ago and resurected recenty.

The implementations is now compatible with Eclipse-MP-1.2 so I think unifying such an API with Eclipse-MP would be possible...
